docs/source/exercises.rst

```diff
@@ -75,6 +75,143 @@ However, things change dramatically when we allow for quantum strategies leading
 
 Can you design a program that simulates this game?
 
+Exercise 5: Quantum Fourier Transform
```

Comment:
is this different from #671

Reply:
My bad, developing off of branches and not being proper about PRs :grimacing: I'll merge that one first.